custom/auth/ante/burntax.go

```diff
@@ -0,0 +1,61 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	treasury "github.com/terra-money/core/x/treasury/types"
+)
+
+// TaxPowerUpgradeHeight is when taxes are allowed to go into effect
+// This will still need a parameter change proposal, but can be activated
+// anytime after this height
+const TaxPowerUpgradeHeight = 9346889
+
+// BurnTaxFeeDecorator will immediately burn the collected Tax
+type BurnTaxFeeDecorator struct {
+	treasuryKeeper TreasuryKeeper
+	bankKeeper     BankKeeper
+}
+
+// NewBurnTaxFeeDecorator returns new tax fee decorator instance
+func NewBurnTaxFeeDecorator(treasuryKeeper TreasuryKeeper, bankKeeper BankKeeper) BurnTaxFeeDecorator {
+	return BurnTaxFeeDecorator{
+		treasuryKeeper: treasuryKeeper,
+		bankKeeper:     bankKeeper,
+	}
+}
+
+// AnteHandle handles msg tax fee checking
+func (btfd BurnTaxFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	// Do not proceed if you are below this block height
+	currHeight := ctx.BlockHeight()
+	if currHeight < TaxPowerUpgradeHeight {
+		return next(ctx, tx, simulate)
+	}
+
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	msgs := feeTx.GetMsgs()
+
+	// At this point we have already run the DeductFees AnteHandler and taken the fees from the sending account
+	// Now we remove the taxes from the gas reward and immediately burn it
+
+	if !simulate {
```

Comment:
you don't seem to have any test with simulate=true, wonder what happens in this case?

Reply:
if the tx is simulated, it does not perform the fee module to burn module transfer for the tax and just moves to the next AnteHandler.  Do you believe there should be a simulate case?  Thanks!

Reply:
OK didn't know that, no sense to add a test in this case then!

Reply:
@simcheolhwan  can you please merge this PR